packages/teleterm/README.md

```diff
@@ -121,6 +122,66 @@ Run `generate-grpc-shared` script from `teleterm/package.json`.
 It generates protobuf files from `*.proto` files in `sharedProcess/api/proto`.
 Resulting files can be found in `sharedProcess/api/protogen`.
 
+## Build process
+
+`yarn package-term` is ran as a part of `yarn build-and-package-term` and is responsible for
+packaging the app code for distribution.
+
+On all platforms, with the exception of production builds on macOS, the `CONNECT_TSH_BIN_PATH` env
+var is used to provide the path to the tsh binary that will be included in the package.
```

Comment:
> Does it end up inside a mock "tsh.app" inside Connect? WDYT about pursuing a setup where the app is exactly the same both for testing and prod?
> 
> Not for this PR, of course, just chatting.

What does testing mean in this case though? By "a mock tsh.app" you mean tsh.app that's not really a bundle but rather just tsh bin under `tsh.app/Contents/MacOS`, like I implemented for environments which don't need Touch ID?

From my own experience, I pretty often do local packaged builds (those would be the "production" builds in Electron terminology if you're not familiar with it) during development to verify that stuff still works there. But usually I do this because of the quirks in Electron behavior between packaged and dev builds.

Having packaged builds to be as close to the real thing as possible is preferable. Using a real tsh.app for those builds though would mean that if any Connect dev wanted to make a regular packaged build, they'd need to have the cert installed on their device, they'd need to provide both `APPLE_USERNAME` & `APPLE_PASSWORD` and compile tsh in a specific way.

We could probably automate away the last part, installing the cert is a one time thing, but providing those env vars seems like a hassle. Pretty sure APPLE_PASSWORD would end up in shell history on someone's computer which is not tragic but not something we want in the end I guess? 😅

But on top of that, it just makes the packaged build much longer because of notarization + signing which I suspect in 95% of cases will not be needed.

We'd also need to solve the issue with email notifications being sent after every notarization (preferably by updating macOS builders).

For now it seems like too much hassle, especially since all of that would be needed only for passwordless login in the end.

But I'm interested to hear other opinions because mine seems to be very, uuh, laissez-faire in that regard and maybe there are better principles to follow there.

---

We could probably come back to this once signing & notarizing is needed for more features.